services/bettersocial/user/BetterSocialUnblockUserV2.js

```diff
@@ -17,11 +28,26 @@ const BetterSocialUnblockUserV2 = async (token, selfUserId, targetUserId) => {
       message: "You can't unblock yourself"
     };
 
+  const targetAnonymousUserId = await UsersFunction.findAnonymousUserId(User, targetUserId);
+  const anonymousBlockFromSign = await isAnonymousUserBlockedFromSignUser(
+    selfUserId,
+    targetAnonymousUserId?.user_id
+  );
   try {
     await sequelize.transaction(async (t) => {
       await UserBlockUserFunction.userUnblock(UserBlockedUser, selfUserId, targetUserId, {
         transaction: t
       });
+
+      if (anonymousBlockFromSign)
+        await UserBlockUserFunction.userUnblock(
+          UserBlockedUser,
+          selfUserId,
+          targetAnonymousUserId?.user_id,
+          {
+            transaction: t
+          }
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this transaction block, you are unblocking the target user and then checking if an anonymous user is blocked from signing in. If they are, you unblock them as well. This code looks fine, but it would be more efficient to check if the anonymous user is blocked before starting the transaction. This way, you can avoid unnecessary transactions when there's no need to unblock the anonymous user.
```diff
- const anonymousBlockFromSign = await isAnonymousUserBlockedFromSignUser(
-   selfUserId,
-   targetAnonymousUserId?.user_id
- );
- try {
-   await sequelize.transaction(async (t) => {
-     await UserBlockUserFunction.userUnblock(UserBlockedUser, selfUserId, targetUserId, {
-       transaction: t
-     });
- 
-    if (anonymousBlockFromSign)
-      await UserBlockUserFunction.userUnblock(
-        UserBlockedUser,
-        selfUserId,
-        targetAnonymousUserId?.user_id,
-        {
-          transaction: t
-        }
-      );
-  });
-} catch (e) {
-  console.log('Error in unblock user v2 sql transaction');
-}
+ try {
+   await UserBlockUserFunction.userUnblock(UserBlockedUser, selfUserId, targetUserId);
+   const anonymousBlockFromSign = await isAnonymousUserBlockedFromSignUser(
+     selfUserId,
+     targetAnonymousUserId?.user_id
+   );
+   if (anonymousBlockFromSign) {
+     await sequelize.transaction(async (t) => {
+       await UserBlockUserFunction.userUnblock(
+         UserBlockedUser,
+         selfUserId,
+         targetAnonymousUserId?.user_id,
+         {
+           transaction: t
+         }
+       );
+     });
+   }
+ } catch (e) {
+   console.log('Error in unblock user v2 sql transaction');
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->